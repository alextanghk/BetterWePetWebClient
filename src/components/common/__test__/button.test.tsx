import React from 'react';
import ReactDOM from 'react-dom'
import { render, cleanup } from '@testing-library/react';
import { BWPButton } from '..';
import renderer from 'react-test-renderer'

afterEach(cleanup)
test("Render without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(<BWPButton />, div)
    ReactDOM.unmountComponentAtNode(div)
})
test('BWP Button rendering correctly in md size', () => {
    const { getByTestId } = render(<BWPButton color="green" size="md">Click me</BWPButton>)
    const testItem = getByTestId("bwp-button")
    expect(testItem).toHaveTextContent("Click me")
    expect(testItem).toHaveClass("btn-bwp-green", "h-10","px-3","rounded-5")
})

test("snapshort of bwp button in md size", () => {
    const snap = renderer.create(<BWPButton color="green" size="md">Click me</BWPButton>).toJSON()
    expect(snap).toMatchSnapshot()
})