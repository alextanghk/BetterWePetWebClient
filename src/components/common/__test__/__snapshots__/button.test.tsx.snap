// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshort of bwp button in md size 1`] = `
<button
  className="h-10 px-3 rounded-5 btn-bwp-green "
  data-testid="bwp-button"
>
  Click me
</button>
`;
